ingest/src/main/java/com/connexta/ingest/adaptors/S3StorageAdaptor.java

```diff
@@ -38,20 +41,31 @@ public S3StorageAdaptor(final S3Client s3Client, final String s3BucketQuarantine
   }
 
   @Override
-  public void store(StoreRequest storeRequest, String key) throws IOException {
+  public void store(StoreRequest storeRequest, String key) throws IOException, StorageException {
     final String fileName = storeRequest.getFileName();
 
     LOGGER.info("Storing {} in bucket \"{}\" with key \"{}\"", fileName, s3BucketQuarantine, key);
-    s3Client.putObject(
-        PutObjectRequest.builder()
-            .bucket(s3BucketQuarantine)
-            .key(key)
-            .contentType(storeRequest.getMimeType())
-            .contentLength(storeRequest.getFileSize())
-            .metadata(ImmutableMap.of("filename", fileName))
-            .build(),
-        RequestBody.fromInputStream(
-            storeRequest.getFile().getInputStream(), storeRequest.getFile().getSize()));
+    try {
+      s3Client.putObject(
+          PutObjectRequest.builder()
+              .bucket(s3BucketQuarantine)
+              .key(key)
+              .contentType(storeRequest.getMimeType())
+              .contentLength(storeRequest.getFileSize())
+              .metadata(ImmutableMap.of("filename", fileName))
+              .build(),
+          RequestBody.fromInputStream(
+              storeRequest.getFile().getInputStream(), storeRequest.getFile().getSize()));
+
+    } catch (SdkServiceException e) {
+      throw new StorageException(
+          "S3 was unable to store " + key + " in bucket " + s3BucketQuarantine, e);
+    } catch (SdkClientException e) {
+      throw new StorageException(
+          "S3 was unavailable and could not store " + key + " in bucket " + s3BucketQuarantine, e);
+    } catch (RuntimeException e) {
+      throw new StorageException("Error storing " + key + " in bucket " + s3BucketQuarantine, e);
```

Comment:
✏️ Could we get a runtime exception from the input stream stuff? Can we make it so that we only catch runtime exception thrown by the .putObject method?

Reply:
The input stream would throw an IO Exception

Reply:
Anything could throw a RuntimeException. I think we'd only want to catch and log the RuntimeException if it comes from the S3Client:
```java
final PutObjectRequest putObjectRequest =
    PutObjectRequest.builder()
        .bucket(s3BucketQuarantine)
        .key(key)
        .contentType(storeRequest.getMimeType())
        .contentLength(storeRequest.getFileSize())
        .metadata(ImmutableMap.of("filename", fileName))
        .build();
final RequestBody requestBody =
    RequestBody.fromInputStream(
        storeRequest.getFile().getInputStream(), storeRequest.getFile().getSize());
try {
  s3Client.putObject(putObjectRequest, requestBody);
} catch (SdkServiceException e) {
  throw new StorageException(
      "S3 was unable to store " + key + " in bucket " + s3BucketQuarantine, e);
} catch (SdkClientException e) {
  throw new StorageException(
      "S3 was unavailable and could not store " + key + " in bucket " + s3BucketQuarantine, e);
} catch (RuntimeException e) {
  throw new StorageException("Error storing " + key + " in bucket " + s3BucketQuarantine, e);
}
```